File: dmn-test-cases/standard/tck/1.1/cl3/0016-some-every/translator/expected/java/dmn/type/TItemPriceImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tItemPrice"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TItemPriceImpl implements TItemPrice {
        private String itemName;
        private java.lang.Number price;

    public TItemPriceImpl() {
    }

    public TItemPriceImpl(String itemName, java.lang.Number price) {
        this.setItemName(itemName);
        this.setPrice(price);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("itemName")
    public String getItemName() {
        return this.itemName;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("itemName")
    public void setItemName(String itemName) {
        this.itemName = itemName;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("price")
    public java.lang.Number getPrice() {
        return this.price;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("price")
    public void setPrice(java.lang.Number price) {
        this.price = price;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0016-some-every/translator/expected/kotlin/dmn/type/TItemPriceImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tItemPrice"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TItemPriceImpl : TItemPrice {
    @get:com.fasterxml.jackson.annotation.JsonGetter("itemName")
    @set:com.fasterxml.jackson.annotation.JsonGetter("itemName")
    override var itemName: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("price")
    @set:com.fasterxml.jackson.annotation.JsonGetter("price")
    override var price: kotlin.Number? = null

    constructor() {
    }

    constructor (itemName: String?, price: kotlin.Number?) {
        this.itemName = itemName
        this.price = price
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
